core/src/main/scala/org/bitcoins/core/wallet/utxo/InternalAddressTag.scala

```diff
@@ -0,0 +1,77 @@
+package org.bitcoins.core.wallet.utxo
+
+/**
+  * An AddressTagType that is native to Bitcoin-S.
+  * InternalAddressTagTypes are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTagType extends AddressTagType
+
+/**
+  * An AddressTag that is native to Bitcoin-S.
+  * InternalAddressTags are still usable when using Bitcoin-S
+  * as a dependency
+  */
+sealed trait InternalAddressTag extends AddressTag
+
+/** An unknown address tag type, should mostly be used for parsing external address tags */
+case class UnknownAddressTagType(name: String) extends InternalAddressTagType
+
+/** An address tag without an unknown type, should mostly be used for parsing external address tags */
+case class UnknownAddressTag(
+    name: String,
+    tagType: AddressTagType = UnknownAddressTagType(""))
```

Comment:
I'm not a fan of this default type, I think users should be forced to be explicit when omitting a tagType